### DISCUSSION TITLE
SEL框架中的time 和 period

### DISCUSSION THREAD

**Discussion by quant1box** [wondertrader/wondertrader#55] [Q&A] [answered]
问题1：在回溯测试SEL 框架的时候，发现策略里面的period 并不能控制策略实际的交易周期， 而是engine.set_sel_strategy(straInfo, time=30, period="min")中的time 能控制k bar合成的周期。请问这二者具体的不同在哪些地方？

问题2：SEL框架 不是在收到所有何有的闭合bar 时才会触发on_calculates事件，是不是就意味着，交易时段不同的商品，只有在二者都在同一个交易时段才会触发计算逻辑？在回测DCE.i.Hot 和CFFEX.IF.Hot 时，由于时间对不齐，同时例如30分钟bar 也存在时间对不齐会在某些时段，有的商品为空值，导致计算失真。

**Comment by wondertrader** [comment, marked as answer]
SEL引擎调度方式分成两种驱动方式：时间和基于行情的时间
当你设定的period是min，这个一定是在交易时间段内进行调度。如你问题中提到的，你可以理解为是m30做一次重算
当你设定的period是day甚至是week，这个时候time就必须是一天中的具体时间，这种就通过本地时间判断做调度，不过这种方式替代方案较多

**Reply by quant1box** [reply]
wondertrader 在这些方面应该是走在市场的前面，市场里面缺乏这样的架构资料，自己也缺乏平台这方面的使用经验，自己做了写尝试，大多是不正确的，希望老板尽快完善相关文档资料或者有时间写些相关的示例程序或者SEL引擎详细一点的使用说明。😁 自己目前期货因子策略，交易周期日线，由于频率不高，使用wt 做持仓同步还能应付，提高频率就完全没办法了，现在只有希望能尽快wt的SEL引擎熟悉起来，能做日内的因子策略，另外感谢大佬提供这么好的交易平台。
